openedx/core/djangoapps/user_api/views.py

```diff
@@ -345,10 +350,63 @@ def post(self, request):
             }
             return JsonResponse(errors, status=400)
 
+        password = data.get("password")
+        password_copy = data.get("password_copy")
+        if password and password_copy and password != password_copy:
+            errors = {
+                'password_copy': [{"user_message": "Passwords don't match."}]
+            }
+            return JsonResponse(errors, status=400)
+
         response = JsonResponse({"success": True})
         set_logged_in_cookies(request, response, user)
+
+        if getattr(settings, 'REGISTRATION_EMAIL_FULL_VERIFICATION', None) is not None:
+            email = data.get("email")
+            error = self._verify_email_really_exists(email)
+            if error:
+                errors = {
+                    'email': [{"user_message": error}]
+                }
+                return JsonResponse(errors, status=400)
+
         return response
 
+    def _verify_email_really_exists(self, email):
+        """Check if a email really exists,
+        returns None if success, otherwise an error message
+        """
+        mx_record = None
+        try:
+            domain = email.rsplit('@', 1)[-1]
+            records = dns.resolver.query(domain, 'MX')
+            mx_record = records[0].exchange
+            mx_record = str(mx_record)
+        except (BaseException, StandardError):
+            return u"Email domain '{domain}' doesn't exist.".format(domain=domain)
+
+        try:
+            # Get local server hostname
+            host = socket.gethostname()
+
+            # SMTP lib setup (use debug level for full output)
+            server = smtplib.SMTP()
+            server.set_debuglevel(0)
+            server.timeout = 10
+
+            # SMTP Conversation
+            server.connect(mx_record)
+            server.helo(host)
+            server.mail(email)
```

Comment:
mail() implements SMTP 'mail' command, it's not icluded in the documentation as a low-lovel method, I found its description here https://hg.python.org/cpython/file/3.5/Lib/smtplib.py